client/ayon_substancepainter/plugins/publish/collect_textureset_images.py

```diff
@@ -209,12 +209,14 @@ def get_export_config(self, instance):
         }
 
         # Create the list of Texture Sets to export.
+        config["exportList"] = []
         export_texture_set = creator_attrs.get("exportTextureSet", [])
-        config["exportList"] = [
-            {"rootPath": texture_set.name()}
-            for texture_set in substance_painter.textureset.all_texture_sets()
-            if not export_texture_set or texture_set.name() in export_texture_set
-        ]
+        if export_texture_set:
+            for export_texture in export_texture_set:
+                config["exportList"].append({"rootPath": export_texture})
+        else:
+            for texture_set in substance_painter.textureset.all_texture_sets():
+                config["exportList"].append({"rootPath": texture_set.name()})
```

Comment:
This is what I meant - because you didn't have the 'else' clause before either.
```suggestion
        if export_texture_set:
            for export_texture in export_texture_set:
                config["exportList"] = [
                    {"rootPath": texture_set_name} 
                    for texture_set_name in export_texture_set
                ]
```


⚠️ 
Also `exportTextureSet` should really be named plural, `exportTextureSets` because it's a list of texture set names, not a single one.

Reply:
but if there is not RootPath, this would mean there is no shader being exported and thus errors out the publisher. We need to figure out how to deal with this issue too.

Reply:
What error? Because my proposal is to use the old logic from prior to this PR. Which means that if it does not work that `develop` branch also does not work?

I think you are referring to passing an empty exportList. But I am saying not passing that at all if no filtering needs to be applied.

Reply:
Exactly, so we change it to not pass the list at all, like before.